src/Microsoft.IdentityModel.Tokens/Encryption/AuthenticatedEncryptionProvider.cs

```diff
@@ -210,6 +224,29 @@ public virtual byte[] Decrypt(byte[] ciphertext, byte[] authenticatedData, byte[
             }
         }
 
+        /// <summary>
+        /// Calls <see cref="Dispose(bool)"/> and <see cref="GC.SuppressFinalize"/>
+        /// </summary>
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        /// <summary>
+        /// Releases managed resources.
+        /// </summary>
+        /// <param name="disposing">true, if called from Dispose(), false, if invoked inside a finalizer.</param>
+        protected virtual void Dispose(bool disposing)
+        {
+            if (!_disposed)
```

Comment:
>if (!_disposed) [](start = 12, length = 15)

nit: why not invert this and return early?

Reply:
@henrik-me I will fix this, thanks.

Reply:
resolved by: https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/commit/a62d951d0c2a5759d03372b404f345b333fcb438